crates/core/src/pattern/ast_node.rs

```diff
@@ -47,27 +53,30 @@ impl ASTNode {
     ) -> Result<Self> {
         let mut args = Vec::new();
         if context.lang.is_comment(sort) {
-            if named_args.len() > 1 {
-                return Err(anyhow!("comment node has more than one field"));
-            }
-            if named_args.is_empty() {
-                return Ok(Self::new(sort, args));
-            }
-            let (name, node) = &named_args[0];
-            if *name != "content" {
-                return Err(anyhow!("unknown field name {} for comment node", name));
+            match named_args.len().cmp(&1) {
```

Comment:
I changed this a little bit to avoid another clone, at which point Clippy suggested I turn it into a `match`. It does look cleaner this way :)